components/files/configurationmanager.hpp

```diff
@@ -63,6 +65,53 @@ struct ConfigurationManager
         bool mSilent;
 };
 
+
+/**
+ * \struct escape_hash_filter
+ */
+struct escape_hash_filter : public boost::iostreams::input_filter
+{
+    static const int sEscape;
+    static const int sHashIdentifier;
+    static const int sEscapeIdentifier;
+
+    escape_hash_filter();
+    virtual ~escape_hash_filter();
+
+    template <typename Source> int get(Source & src);
+
+    private:
+        std::queue<int> mNext;
+        int mPrevious;
+
+        bool mSeenNonWhitespace;
+        bool mFinishLine;
+};
+
+/**
+ * \class EscapeHashString
+ */
+class EscapeHashString : public std::string
+{
+    private:
+        std::string mData;
+    public:
+        static std::string processString(const std::string & str);
+        static std::vector<std::string> toStdStringVector(const std::vector<EscapeHashString> & vec);
+
+        EscapeHashString();
+        EscapeHashString(const std::string & str);
+        EscapeHashString(const std::string & str, size_t pos, size_t len = std::string::npos);
+        EscapeHashString(const char * s);
+        EscapeHashString(const char * s, size_t n);
+        EscapeHashString(size_t n, char c);
+        template <class InputIterator>
+        EscapeHashString(InputIterator first, InputIterator last);
+
+        std::string toStdString() const;
+        std::string * toStdStringPtr() const;
```

Comment:
It can probably go - I was having an issue with namespaces, but it looked like it was actually with types, so I was double checking something, and then didn't clean it up. This should be resolved in a moment.